### ISSUE TITLE
Remove Python 2 references from extract_1d

### ISSUE THREAD

**Issue by philhodge** [STScI-JWST/jwst#1502]
The `round` function was used in two places in function `_extract_colpix` in extract1d.py.  About a year ago I added a `p2_round` function that was supposed to give the same result as the Python 2 `round` function, for either Python 2 or Python 3.  `p2_round` does actually work for positive values of its argument, but for some negative arguments it is off by one.  Instead of trying to emulate Python 2 behavior of `round`, just use `math.floor(x + 0.5)`.

**Comment by mcara** [comment]
[`drizzlepac.wcs_functions.py`](https://github.com/spacetelescope/drizzlepac/blob/master/lib/drizzlepac/wcs_functions.py#L540) has a function `_py2round()` that implements Python 2-style `round()` logic.

```python
def _py2round(x):
    """
    This function returns a rounded up value of the argument, similar
    to Python 2.
    """
    if hasattr(x, '__iter__'):
        rx = np.empty_like(x)
        m = x >= 0.0
        rx[m] = np.floor(x[m] + 0.5)
        m = np.logical_not(m)
        rx[m] = np.ceil(x[m] - 0.5)
        return rx

    else:
        if x >= 0.0:
            return np.floor(x + 0.5)
        else:
            return np.ceil(x - 0.5)
```

It hope it works as expected (with array support too) - please test it.

I think using Python 3 here is the worst option.

**Comment by philhodge** [comment]
The drizzlepac _py2round function does give the Python 2 behavior of round, for the cases I tested (near -1.5, near 0, and near +1.5; where "near" means value - 0.05 to value + 0.05 in steps of 0.01).

It seems to me, however, that the Python 2 round function is not actually what is needed in _extract_colpix.  Do you really want to round away from 0 for negative arguments?  I do agree that the Python 3 round function is out of the question.  However, it seems to me that floor(x + 0.5) does give the value we want.

**Comment by mcara** [comment]
I think I agree with you about replacing rounding with `np.floor(x+0.5)`.

Looking at the code, it seems that `i1` and `i2` in https://github.com/STScI-JWST/jwst/blob/b262a2c7b8acf24deea1614c5d0ebd63d8ff61e6/jwst/extract_1d/extract1d.py#L357 should not be smaller than `-0.5` ever. In particular `i2`, in principle, should be strictly larger than `-0.5`. Also `x[1]` in https://github.com/STScI-JWST/jwst/blob/b262a2c7b8acf24deea1614c5d0ebd63d8ff61e6/jwst/extract_1d/extract1d.py#L342 should be strictly larger than `-0.5`. In any case, when `ii2` is computed as `ii2 = min(ns, int(p2_round(i2)))`, it seems that it would not be affected by how rounding of `-0.5` is done since `ii2>ns>=0`.

So, it seems to me that even your current implementation of `p2_round()` should not be causing any issues but it also appears that it can be safely replaced with `np.floor(x+0.5)`.

**Comment by philhodge** [comment]
I agree that replacing `p2_round(x)` with `math.floor(x + 0.5)` should have no effect on the results.  The function `p2_round` is a "reference" to Python 2, however, so I wanted to remove it.  Also, I think the code is clearer when it uses the `math.floor` function.

**Comment by hbushouse** [comment]
Fixed by #1519
